fastdds_python/test/api/test_publisher.py

```diff
@@ -61,6 +66,330 @@ def test_get_set_qos():
            factory.delete_participant(participant))
 
 
+def test_get_set_listener():
+    """
+    This test checks:
+    - Publisher::get_listener
+    - Publisher::set_listener
+    - Publisher::get_status_mask
+    - StatusMask::operator ==
+    - StatusMask::operator <<
+    """
+    factory = fastdds.DomainParticipantFactory.get_instance()
+    assert(factory is not None)
+    participant = factory.create_participant(
+            0, fastdds.PARTICIPANT_QOS_DEFAULT)
+    assert(participant is not None)
+    publisher = participant.create_publisher(fastdds.PUBLISHER_QOS_DEFAULT)
+    assert(publisher is not None)
+
+    # Overload 1
+    listener = PublisherListener()
+    assert(listener is not None)
+    assert(fastdds.ReturnCode_t.RETCODE_OK ==
+           publisher.set_listener(listener))
+    assert(publisher.get_listener() == listener)
+    assert(fastdds.StatusMask.all() == publisher.get_status_mask())
+
+    # Overload 2
+    # - StatusMask.none
+    listener = PublisherListener()
+    assert(listener is not None)
+    assert(fastdds.ReturnCode_t.RETCODE_OK ==
+           publisher.set_listener(
+               listener, fastdds.StatusMask.none()))
+    assert(publisher.get_listener() == listener)
+    assert(fastdds.StatusMask.none() == publisher.get_status_mask())
```

Comment:
Similar to @MiguelCompany 's concerns in previous reviews, this code block is repeated with minor changes. Maybe it could be changed into a function.

Same for the others below